### PR TITLE
Add OnBlur to SearchInput

### PR DESCRIPTION
**Pull request by fleonard**
## Screenshot / video

[if applicable for UI changes]

## What does this do?

[A short description of what the PR changes and why it's necessary]

- Give the reviewers context
- Is the impact global or localised?
- Any tech debt created or discovered?
- Include links to Jira ticket and Figma file if applicable

### CODE REVIEW

File: src/SearchInput/SearchInput.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ export interface SearchInputProps extends CommonFieldTypes {
   onFound: (element: string) => void
   resultsRelativePosition?: boolean
   showIcon?: boolean
+  onBlur?: (e: FormEvent<HTMLInputElement>) => void
```
Comment (mickaelzhang) [inline src/SearchInput/SearchInput.tsx:22]: I think the right typing would be:
```suggestion
  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
```

File: src/SearchInput/SearchInput.tsx [hunk 2]
```diff
@@ -92,6 +94,9 @@ export const SearchInput: FC<SearchInputProps> = ({
             onChange={updateInputState}
             outlined={outlined}
             selected={selected}
+            onBlur={(e) => {
+              onBlur && onBlur(e)
+            }}
```
Comment (mickaelzhang) [inline src/SearchInput/SearchInput.tsx:99]:
```suggestion
            onBlur={onBlur}
```

File: src/SearchInput/SearchInput.tsx (hunk 1 above)
Comment (fleonard) [reply src/SearchInput/SearchInput.tsx:22]: We are using form event everywhere else tho 🤔

Comment (fleonard) [reply src/SearchInput/SearchInput.tsx:22]: Should I change it across all instances?

Comment (fleonard) [reply src/SearchInput/SearchInput.tsx:22]: MDN actually says it's a FocusEvent so I'll update it

**Review by githab666 — APPROVED** [review]

File: src/SearchInput/SearchInput.tsx (hunk 1 above)
Comment (mickaelzhang) [reply src/SearchInput/SearchInput.tsx:22]: The typing is probably wrong yes, I believe for `onFocus` and `onBlur`, it should be a `FocusEvent`

**Review by mickaelzhang — DISMISSED** [review]

**Comment by fleonard** [comment]
@mickaelzhang I've realised I've never updated the Text Readme with the new typo so I've done it now

**Review by mickaelzhang — APPROVED** [review]
@fleonard I'm wondering if it is used? I never needed it on my end

**Pull request merged**